python-avd/pyavd/_eos_designs/structured_config/network_services/application_traffic_recognition.py

```diff
@@ -22,17 +21,184 @@ class ApplicationTrafficRecognitionMixin(Protocol):
+    ) -> None:
+        """
+        Update atr with the prefix set.
+
+        Default Application Profile for control plane uses prefix sets.
+
+        Args:
+            atr: The ApplicationTrafficRecognition object local to the module to track the set objects
+            prefix_set_name: Name of the prefix set to configure
+            application_name: Name of the application to use in error message
+            is_control_plane_vt: Indicates if we are setting up the ApplicationTrafficRecognition for Control-Plane Virtual Topology
+
+        Raises:
+            AristaAvdInvalidInputsError: if the prefix-set name is not the appropriate control-plane one and is not present in the inputs.
+        """
+        if prefix_set_name in self.inputs.application_classification.field_sets.ipv4_prefixes:
+            atr.field_sets.ipv4_prefixes.append(self.inputs.application_classification.field_sets.ipv4_prefixes[prefix_set_name])
+        elif is_control_plane_vt and (
+            (self.shared_utils.is_wan_client and prefix_set_name == self._wan_cp_app_dst_prefix)
+            or (self.shared_utils.is_wan_server and prefix_set_name == self._wan_cp_app_src_prefix)
+        ):
+            # use default prefix-set
+            atr.field_sets.ipv4_prefixes.append(self.default_control_plane_prefix_set())
+        else:
+            msg = (
+                f"The IPv4 prefix field set {prefix_set_name} used in the application {application_name} "
+                "is undefined in 'application_classification.fields_sets.ipv4_prefixes'."
+            )
+            raise AristaAvdInvalidInputsError(msg)
+
+    def _update_port_set(
+        self: AvdStructuredConfigNetworkServicesProtocol, atr: EosCliConfigGen.ApplicationTrafficRecognition, port_set_name: str, application_name: str
+    ) -> None:
+        """
+        Update atr with the port set.
+
+        Args:
+            atr: The ApplicationTrafficRecognition object local to the module to track the set objects
+            port_set_name: Name of the port set to configure
+            application_name: Name of the application to use in error message
 
-        self._set_control_plane_application_profile()
+        Raises:
+            AristaAvdInvalidInputsError: if the port-set name is not present in the inputs.
+        """
+        if port_set_name not in self.inputs.application_classification.field_sets.l4_ports:
+            msg = (
+                f"The L4 Ports field set {port_set_name} used in the application {application_name} "
+                "is undefined in 'application_classification.fields_sets.l4_ports'."
+            )
+            raise AristaAvdInvalidInputsError(msg)
+
+        atr.field_sets.l4_ports.append(self.inputs.application_classification.field_sets.l4_ports[port_set_name])
 
-    #  self._wan_control_plane_application_profile is defined in utils.py
     @cached_property
     def _wan_control_plane_application(self: AvdStructuredConfigNetworkServicesProtocol) -> str:
```

Comment:
IMO we should remove these. Use a constant if you wish. If we ever make it configurable it will be a schema field anyway.